src/miner.cpp

```diff
@@ -473,6 +518,137 @@ void BlockAssembler::addScoreTxs(std::vector<const CTxMemPoolEntry *> *vtxe)
+// having transactions with all the same fees and Two, the typical child pays for parent scenario has only
+// two transactions with the child having the higher fee. And neither of these two types of packages could
+// cause any loss of fees with this mining algorithm, when the block is nearly full.
+void BlockAssembler::addPackageTxs(std::vector<const CTxMemPoolEntry *> *vtxe, bool fCanonical)
+{
+    AssertLockHeld(mempool.cs);
+
+    CTxMemPool::txiter iter;
+    uint64_t nPackageFailures = 0;
+    for (auto mi = mempool.mapTx.get<ancestor_score>().begin(); mi != mempool.mapTx.get<ancestor_score>().end(); mi++)
+    {
+        iter = mempool.mapTx.project<0>(mi);
+
+        uint64_t packageSize = iter->GetSizeWithAncestors();
+        CAmount packageFees = iter->GetModFeesWithAncestors();
+        unsigned int packageSigOps = iter->GetSigOpCountWithAncestors();
+
+        // Skip txns we know are in the block
+        if (SkipMapTxEntry(iter))
+        {
+            continue;
+        }
+
+        // Get any unconfirmed ancestors of this txn
+        CTxMemPool::setEntries ancestors;
+        uint64_t nNoLimit = std::numeric_limits<uint64_t>::max();
+        std::string dummy;
+        mempool._CalculateMemPoolAncestors(
+            *iter, ancestors, nNoLimit, nNoLimit, nNoLimit, nNoLimit, dummy, &inBlock, false);
+
+        // Include in the package the current txn we're working with
+        ancestors.insert(iter);
+
+        // Recalculate sigops and package size, only if there were txns already in the block for
+        // this set of ancestors
+        if (iter->GetCountWithAncestors() > ancestors.size())
+        {
+            packageSize = 0;
+            packageSigOps = 0;
+            for (auto &it : ancestors)
+            {
+                packageSize += it->GetTxSize();
+                packageSigOps += it->GetSigOpCount();
+            }
+        }
+        if (packageFees < ::minRelayTxFee.GetFee(packageSize) && nBlockSize >= nBlockMinSize)
+        {
+            // Everything else we might consider has a lower fee rate so no need to continue
+            return;
+        }
+
+        // Test if package fits in the block
+        if (nBlockSize + packageSize > nBlockMaxSize)
+        {
+            if (nBlockSize > nBlockMaxSize * .50)
+            {
+                nPackageFailures++;
+            }
+
+            // If we keep failing then the block must be almost full so bail out here.
+            if (nPackageFailures >= MAX_PACKAGE_FAILURES)
+                return;
+            else
+                continue;
+        }
+
+        // Test that the package does not exceed sigops limits
+        if (!TestPackageSigOps(packageSize, packageSigOps))
+        {
+            continue;
+        }
+        // Test if all tx's are Final
+        if (!TestPackageFinality(ancestors))
+        {
+            continue;
+        }
+
+        // The Package can now be added to the block.
```

Comment:
Good point but we can't because we still have both in our regression tests such as ctor.py and also in the miner we later either do the sort or not based on the "canonical" flag.  I think if we wan't to remove support for TTOR then it should be done in a different PR.